server/middlewares/UserValidation.js

```diff
@@ -6,6 +6,20 @@ const message = [];
  * @description Helps perform validations on user request body.
  */
 class UserValidation {
+  /**
+    * @description - This method checks if a user enters an email.
+    * @param {object} req - The request object bearing the email.
+    * @param {object} res - The response object sent to the next middleware.
+    * @param {object} next - The callback function to the next middleware.
+    * @returns {object} - The error object with message.
+    * @memberOf UserValidation
+    * @static
+    */
+  static validateUserLogin(req, res, next) {
+    UserValidation.validateEmail(req);
+    UserValidation.sendFormattedError(req, res, next);
+  }
```

Comment:
I think we should be validating `fullName` here, you could add a method `validateFullName` that does this if the `fullName` is provided(on signUp).

Nice work on refactoring all this.